### ISSUE TITLE
`no-unsafe-enum-comparison` misses string literal union types

### ISSUE THREAD

**Issue by JasonKaz** [oxc-project/tsgolint#934]
Disclosure: Generated by AI, but manually tested.

Using `oxlint@1.62.0` and `oxlint-tsgolint@0.22.1`.

```ts
/**
 * Reproduction: tsgolint `no-unsafe-enum-comparison` misses string literal union types
 *
 * Run: npx oxlint --type-aware .tmp-repro/no-unsafe-enum-comparison-repro.ts
 *
 * Expected: 3 errors (cases A, B, C)
 * Actual:   1 error  (only case A)
 */

enum UXSortDir {
    asc = 'asc',
    desc = 'desc',
}

type SortDir = 'asc' | 'desc'; // string literal union — same values as the enum

// ─── CASE A: plain `string` — CORRECTLY flagged ──────────────────────────────
// @typescript-eslint/no-unsafe-enum-comparison: ✅ flagged by tsgolint
function compareWithString(sortDir: string) {
    return sortDir === UXSortDir.asc;
}

// ─── CASE B: string literal union — MISSED by tsgolint ───────────────────────
// @typescript-eslint/no-unsafe-enum-comparison: ❌ NOT flagged (expected: error)
function compareWithUnion(sortDir: SortDir) {
    return sortDir === UXSortDir.asc;
}

// ─── CASE C: inline string literal union — MISSED by tsgolint ────────────────
// @typescript-eslint/no-unsafe-enum-comparison: ❌ NOT flagged (expected: error)
function compareWithInlineUnion(sortDir: 'asc' | 'desc') {
    return sortDir === UXSortDir.asc;
}

// ─── CASE D: same enum type — correctly silent ───────────────────────────────
// Both sides are UXSortDir, so comparison is safe — no error expected
function compareEnumToEnum(sortDir: UXSortDir) {
    return sortDir === UXSortDir.asc;
}

// Prevent no-unused-vars noise
compareWithString('asc');
compareWithUnion('asc');
compareWithInlineUnion('asc');
compareEnumToEnum(UXSortDir.asc);
```
